juliasamartseva/Contacts/Sources/Contacts/Contacts.swift

```diff
@@ -101,36 +171,102 @@ public class ContactList {
   public func update(contact: Contact) throws {
     if let index = contacts.firstIndex(where: { $0.id == contact.id }) {
       contacts[index] = contact
-      try writeToFile()
+      try writeToFile(.contacts)
       logger.info("Successfully changed contact with id \(contact.id) to new contact")
     } else {
       logger.info("No contact with \(contact.id) was found")
     }
   }
+  
+  public func add(group: ContactGroup, to contact: Contact) throws {
+    /// The contact with the given id must be present in the ContactList
+    /// The group must be present in the group list. Otherwise, it can be added to the group list via add(group) function
+    if let index = contacts.firstIndex(where: { $0.id == contact.id }),
+      groups.contains(group)
+    {
+      contacts[index].add(group: group)
+      try writeToFile(.contacts)
+      logger.info("Successfully added group \(group.id) to contact with id \(contact.id)")
+    } else {
+      logger.info("No contact with \(contact.id) was found or the group with \(group.id) doesn't exist")
+    }
+  }
+
+  public func remove(group: ContactGroup, from contact: Contact) throws {
+    if let index = contacts.firstIndex(where: { $0.id == contact.id }),
+      groups.contains(group)
+    {
+      contacts[index].remove(group: group)
+      try writeToFile(.contacts)
+      logger.info("Successfully added group \(group.id) to contact with id \(contact.id)")
+    } else {
+      logger.info("No contact with \(contact.id) was found or the group with \(group.id) doesn't exist")
+    }
```

Comment:
I would invert a little bit
```suggestion
    guard let index = contacts.firstIndex(where: { $0.id == contact.id }),
      groups.contains(group) 
    else {
       logger.info("No contact with \(contact.id) was found or the group with \(group.id) doesn't exist")
    }
    contacts[index].remove(group: group)
    try writeToFile(.contacts)
    logger.info("Successfully added group \(group.id) to contact with id \(contact.id)")
  
```
here and in the next code block